current/runtime/fees.md

```diff
@@ -4,97 +4,32 @@ lang: en
 title: Transaction Fees
 ---
 
-When transactions are submitted to a blockchain, they are executed by the nodes in the network. To
-be economically sustainable, nodes charge a fee to execute a transaction. This fee must be covered
-by the sender of the transaction. The cost to execute transactions can vary, so Substrate provides a
-flexible mechanism to characterize the minimum cost to include a transaction in a block.
+When transactions or other extrinsics are submitted to a blockchain, they are dispatched (invoked)
+by the nodes in the network. To be economically sustainable, nodes charge a fee to dispatch an
+extrinsic. This fee must be covered by the sender of the extrinsic. Since the resources needed to
+execute an extrinsic can vary, Substrate provides a flexible mechanism called "weights" to
+characterize the _time_ it takes to execute an extrinsic.
```

Comment:
```suggestion
When a block author constructs a block, it must limit the block's execution time. A block body
consists of a series of [extrinsics](/current/learn-substrate/extrinsics.md). Since the resources
needed to execute an extrinsic can vary, Substrate provides a flexible mechanism called "weights"
to characterize the _time_ it takes to execute an extrinsic. To be economically sustainable and to
limit spam, some transactions --- primarily those dispatched by users --- require a fee prior to
transaction execution.
```